### ISSUE TITLE
[BUG] Incorrect display of negative signs in tick labels with non-default fonts in PDF outputs

### ISSUE THREAD

**Issue by singularitti** [JuliaPlots/Plots.jl#4929]
## Details

When using any non-default [fonts available in the GR framework](https://gr-framework.org/fonts.html) with the `plot` function, the negative signs (`-`) in the tick labels on both the x and y axes are incorrectly displayed as question marks (`?`). This issue occurs specifically when saving plots as PDF files. Saving the same plots as PNG files shows the negative signs correctly, indicating that the issue is isolated to PDF outputs.

### Steps to Reproduce

1. Install and load the `Plots` package.
2. Set `gr()` as the backend.
3. Execute the following plotting commands using various non-default fonts:
   ```julia
   using Plots
   gr()
   plot(-4:4, -5:3; fontfamily="Palatino Roman")  # Change font as needed
   xlabel!("-x")
   ylabel!("-y")
   savefig("x.pdf")  # Save as PDF and observe the issue
   savefig("x.png")  # Save as PNG for comparison
   ```
4. Open both `x.pdf` and `x.png` to view the results.

![PDF](https://github.com/JuliaPlots/Plots.jl/assets/25192197/9fada924-9849-4ce4-9d63-3eaf07b81101)

![PNG](https://github.com/JuliaPlots/Plots.jl/assets/25192197/73112e5f-5ef7-49e0-b77b-036247519f2e)

### Expected Result

The tick labels with negative numbers should correctly display the negative sign in both PDF and PNG outputs.

### Actual Result

In PDF outputs, the negative signs in the tick labels are replaced with question marks (`?`). In PNG outputs, the negative signs are displayed correctly.

### Backends

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |  x   |     |
pythonplot   |     |     |
plotlyjs     |     |     |
pgfplotsx    |     |     |
unicodeplots |     |     |
inspectdr    |     |     |
gaston       |     |     |

### Versions

Plots.jl version: `v1.40.4`
Backend  version (`]st -m <backend(s)>`): `v0.73.5`
Output of `versioninfo()`:
```
Julia Version 1.10.3
Commit 0b4590a5507 (2024-04-30 10:59 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 10 × Apple M1 Pro
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
```

**Comment by luboshanus** [comment]
Hi, I would like to ask, if there is any progress with this bug when saving a figure to PDF with other fonts? It still renders it wrongly.

**Comment by emima94** [comment]
If you use the very similar fontfamily "Times New Roman" it seems to work :)

using Plots
gr()
plot(-4:4, -5:3; fontfamily="Palatino Roman")  # Change font as needed
xlabel!("-x")
ylabel!("-y")
savefig("x1.pdf")  # Save as PDF and observe the issue
savefig("x1.png")  # Save as PNG for comparison

plot(-4:4, -5:3; fontfamily="Times New Roman")  # Change font as needed
xlabel!("-x")
ylabel!("-y")
savefig("x2.pdf")  # Save as PDF and observe the issue
savefig("x2.png")  # Save as PNG for comparison

**Comment by luboshanus** [comment]
Yes, but that is not a solution. Why it works differently for different fonts? **It renders each font differently, why?** 

You can find a way to do Palatino too (different representations (character) of sign before a number) but it is scratching by right hand behind left ear.
